smarts/sstudio/generators.py

```diff
@@ -289,6 +296,38 @@ def _writexml(
                 )
             )
 
+    def write_trip_xml(self, traffic, doc, fill_in_gaps):
+        """Wrtes a traffic spec into a route file. Typically this would be the source
+        data to Sumo's DUAROUTER.
+        """
+        # Make sure all routes are "resolved" (e.g. `RandomRoute` are converted to
+        # `Route`) so that we can write them all to file.
+        resolved_routes = {}
+        for route in {trip.route for trip in traffic.trips}:
+            resolved_routes[route] = self.resolve_route(route, fill_in_gaps)
+
+        for route in set(resolved_routes.values()):
+            doc.stag("route", id=route.id + "trip", edges=" ".join(route.roads))
+
+            # We don't de-dup flows since defining the same flow multiple times should
+            # create multiple traffic flows. Since IDs can't be reused, we also unique
+            # them here.
+        for trip_idx, trip in enumerate(traffic.trips):
+            route = resolved_routes[trip.route]
+            actor = trip.actor
+            doc.stag(
+                "vehicle",
+                id="{}".format(trip.vehicle_name),
```

Comment:
The one thing we should do is ensure that the vehicle ids are unique by giving an error if there are multiple trips with the same vehicle name.